ui_framework/components/tabs/tabs.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { KuiTab } from './tab';
+
+export const KuiTabs = ({
+  children,
+  selectedTabIndex,
+  onSelectedTabChanged,
+  className,
+  ...rest
+}) => {
+  const classes = classNames('kuiTabs', className);
+  const tabs = children.map((child,index)=>(
+      <KuiTab
+        key={index}
+        onClick={() => onSelectedTabChanged(index)}
+        isSelected={index === selectedTabIndex}
+      >
+        {child}
+      </KuiTab>
+    ));
+
+  return (
+    <div
+      className={classes}
+      {...rest}
+    >
+      {tabs}
+    </div>
+  );
+};
+
+const selectedTabIndexCheck = (props, propName, componentName) => {
+  const childrenLength = props.children.length;
+  const selectedTabIndex = props.selectedTabIndex;
+
+  if (childrenLength > 0) {
+    if ((selectedTabIndex !== undefined) && (selectedTabIndex < 0) || (selectedTabIndex > childrenLength - 1)) {
+      return new Error(`${componentName}'s selectedTabIndex must be within 0 and tabs count - 1.`);
+    }
+  } else {
+    if (selectedTabIndex !== undefined) {
+      return new Error(`${componentName}'s selectedTabIndex must be undefined if there is no tab to select.`);
+    }
+  }
+};
```

Comment:
Can we refactor this to use the "early exit" pattern, to simplify the conditions we're testing against?

```javascript
// An undefined selected tab is OK in all situations.
if (selectedTabIndex === undefined) {
  return;
}

if (childrenLength === 0) {
  throw new Error(`${componentName}'s selectedTabIndex must be undefined if there is no tab to select.`);
}

if (selectedTabIndex < 0 || selectedTabIndex > childrenLength -1) {
  throw new Error(`${componentName}'s selectedTabIndex must be within the range defined by the number of tabs.`);
}
```